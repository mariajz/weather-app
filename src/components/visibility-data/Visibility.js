import React from 'react';
import WeatherDetailsCard from '../../commons/weather-details-card';
import { getVisibilityText } from '../../utils/helpers';

const Visibility = ({ visibilityProps }) => {
    const { visibility } = visibilityProps;
    const { visibilityText } = getVisibilityText(visibility);
    const feelsLikeCardProps = {
        iconName: 'visibility',
        cardName: 'VISIBILITY',
        title: `${visibility} km`,
        subtitle: '',
        content: visibilityText,
    };

    return <WeatherDetailsCard weatherDetailsCardProps={feelsLikeCardProps} />;
};

export default Visibility;
