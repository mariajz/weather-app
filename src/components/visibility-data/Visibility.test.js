import { render } from '@testing-library/react-native';
import React from 'react';
import Visibility from './Visibility';

const defaultProps = {
    visibilityProps: {
        visibility: 17,
    },
};

describe('Visibility', () => {
    it('should render Visibility', () => {
        const container = render(<Visibility {...defaultProps} />);

        expect(container).toMatchSnapshot();
    });
});
