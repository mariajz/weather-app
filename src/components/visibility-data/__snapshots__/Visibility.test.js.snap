// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visibility should render Visibility 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(255,255,255, 0.2)",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "height": 170,
      "marginTop": 10,
      "paddingBottom": 20,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
      "width": "49%",
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <Image
      size={15}
      source={
        {
          "testUri": "../../../assets/icons/visibility.png",
        }
      }
      style={
        {
          "height": 15,
          "width": 15,
        }
      }
      testID="icon"
      tintColor="white"
    />
    <Text
      marginLeft={5}
      size={14}
      style={
        {
          "color": "white",
          "fontSize": 14,
          "marginLeft": 5,
        }
      }
    >
      VISIBILITY
    </Text>
  </View>
  <View
    style={
      {
        "height": 120,
        "justifyContent": "space-between",
        "marginTop": 8,
      }
    }
  >
    <View>
      <Text
        size={28}
        style={
          {
            "color": "white",
            "fontSize": 28,
            "marginLeft": 0,
          }
        }
      >
        17 km
      </Text>
      <Text
        size={20}
        style={
          {
            "color": "white",
            "fontSize": 20,
            "marginLeft": 0,
          }
        }
      />
    </View>
    <View />
    <Text
      size={10}
      style={
        {
          "color": "white",
          "fontSize": 10,
          "marginLeft": 0,
        }
      }
    >
      It's perfectly clear right now.
    </Text>
  </View>
</View>
`;
